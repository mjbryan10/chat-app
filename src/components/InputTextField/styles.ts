import styled, {ThemeProps, Theme} from 'styled-components';

export const Input = styled.input`
   background-color: ${({theme: {background}}: ThemeProps<Theme>) =>
      background.inputField ? background.inputField : '#6C757D'};
   padding: 1em;
   color: ${ ({theme: {color}}: ThemeProps<Theme>) => color.text || 'inherit'};
`;
