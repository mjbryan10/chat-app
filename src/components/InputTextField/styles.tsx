import styled from 'styled-components';

export const Input = styled.input`
   background-color: ${({theme}) =>
      theme.inputBackground ? theme.inputBackground : '#6C757D'};
   padding: 1em;
   color: inherit;
`;
