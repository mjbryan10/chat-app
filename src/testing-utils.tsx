import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }