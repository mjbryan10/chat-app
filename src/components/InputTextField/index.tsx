import React, { useState, FC } from 'react';
import * as S from './styles';

export interface Props {
   disabled?: boolean;
   placeholder?: string;
   initialValue?: string;
   handleChange?: (value: string) => void;
}

const InputTextField: FC<Props> = ({
   disabled = false,
   placeholder = '',
   initialValue = '',
   handleChange,
}) => {
   const [value, setValue] = useState(initialValue);

   const onChange = (event: React.SyntheticEvent<HTMLInputElement>) => {
      event.preventDefault();
      const { value } = event.currentTarget;
      if (disabled) return;
      setValue(value);
      if (handleChange) handleChange(value);
   };
   return (
      <S.Input
         value={value}
         onChange={onChange}
         placeholder={placeholder}
         data-testid="input-text-field"
      />
   );
};

export default InputTextField;
