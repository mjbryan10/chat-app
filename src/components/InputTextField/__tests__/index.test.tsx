import React from 'react';
import { render, fireEvent } from 'testing-utils';
import InputTextField from '../../InputTextField';

describe('InputTextField Component', () => {
   test('should update local state on user input change', () => {
      const {getByTestId} = render(<InputTextField initialValue={'intial text'} />);
      const input = getByTestId('input-text-field') as HTMLInputElement;

      fireEvent.change(input, { target: { value: 'Changed text' } });

      expect(input.value).toBe('Changed text');
   });

   test('should not update local state on user input change when disabled', () => {
      const {getByTestId} = render(<InputTextField initialValue={'Unchanged text'} disabled />);
      const input = getByTestId('input-text-field') as HTMLInputElement;

      fireEvent.change(input, { target: { value: 'A changed text value' } });

      expect(input.value).toBe('Unchanged text');
   });

   test('should trigger change handler if one is passed down', () => {
      const testHandler = jest.fn();
      const {getByTestId} = render(
         <InputTextField initialValue={'Unchanged text'} handleChange={testHandler} />
      );
      const input = getByTestId('input-text-field') as HTMLInputElement;

      fireEvent.change(input, { target: { value: 'A changed text value' } });

      expect(testHandler).toBeCalled();
      expect(testHandler).toBeCalledWith('A changed text value');
   });

   test('should render correctly', () => {
      const { asFragment } = render(<InputTextField />);
   
      expect(asFragment()).toMatchSnapshot();
   });
});
