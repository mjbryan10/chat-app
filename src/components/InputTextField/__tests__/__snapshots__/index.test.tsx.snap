// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextField Component should render correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm kuGIAZ"
    data-testid="input-text-field"
    placeholder=""
    value=""
  />
</DocumentFragment>
`;
